jwst/datamodels/model_base.py

```diff
@@ -88,22 +96,13 @@ def __init__(self, init=None, schema=None, extensions=None,
         self._strict_validation = self.get_envar("STRICT_VALIDATION",
                                                  strict_validation)
 
-        # Construct the path to the schema files
-        filename = os.path.abspath(inspect.getfile(self.__class__))
-        base_url = os.path.join(
-            os.path.dirname(filename), 'schemas', '')
-
         # Load the schema files
         if schema is None:
-            schema_path = os.path.join(base_url, self.schema_url)
+            schema_path = os.path.join(URL_PREFIX, self.schema_url)
             # Create an AsdfFile so we can use its resolver for loading schemas
-            asdf_file = AsdfFile(extensions=self._extensions)
-            if hasattr(asdf_file, 'resolver'):
-                file_resolver = asdf_file.resolver
-            else:
-                file_resolver = self.get_resolver(asdf_file)
+            asdf_file = AsdfFile()
             schema = asdf_schema.load_schema(schema_path,
-                                             resolver=file_resolver,
+                                             resolver=asdf_file.resolver,
```

Comment:
Just a note but I think it would be useful to eventually provide an API in ASDF that does not require the construction of an `AsdfFile` object.